python/ray/utils.py

```diff
@@ -614,3 +614,48 @@ def try_to_create_directory(directory_path, warn_if_exist=True):
     # Change the log directory permissions so others can use it. This is
     # important when multiple people are using the same machine.
     try_make_directory_shared(directory_path)
+
+
+def try_to_symlink_directory(directory_path, symlink_path):
+    """Attempt to create a symlink to an existing directory.
+    If the directory doesn't exist, the symlink path exists, or the symlink
+    failed to be created, a warning will be logged and the symlink will not
+    be created. If a symlink exists in the path, it will be attempted to be
+    removed and replaced.
+
+    Args:
+        directory_path: The path of the existing directory.
+        symlink_path: The path of the symlink to create.
+    """
+    logger = logging.getLogger("ray")
+    directory_path = os.path.expanduser(directory_path)
+    symlink_path = os.path.expanduser(symlink_path)
```

Comment:
FYI the canonical way to do this is to create the symlink as a tmp file, and then do an atomic move.